pkg/provider/configure_clients.go

```diff
@@ -172,6 +179,20 @@ func createOnCallClient(providerConfig ProviderConfig) (*onCallAPI.Client, error
 	return onCallAPI.New(providerConfig.OncallURL.ValueString(), providerConfig.OncallAccessToken.ValueString())
 }
 
+func createConnectionsClient(client *common.Client, providerConfig ProviderConfig) error {
```

Comment:
We should add support for custom headers to be injected, such as`Grafana-Terraform-Provider`. As an example: https://github.com/grafana/terraform-provider-grafana/pull/1828/files

Reply:
Added in 7ff3e9c0e6bab52aa90dd2b22bbbf9f13900696e